Adding a section on retrieving all documents with paging (#158)
stargate/docs: modules/developers-guide/pages/document-using.adoc

The `page-size` parameter is included to get all the documents, rather than the
last inserted document.

==== Retrieving all documents with Paging
With a collection of many documents it can be helpful to page through only
a subset of the results at a time. For this example, we will use a `page-size`
of 5. Start by sending a `GET` request to
`/v2/namespaces/{namespace_name}/collections/{collections_name}page-size=5` to
retrieve first page of documents in the collection:

[tabs]
====
cURL command (/v2)::
+
--
[source,shell]
----
include::example$docapi/curl_get_all_docs_w_paging_1.sh[]
----
--

Result::
+
--
[source,plaintext, subs="attributes+"]
----
include::example$result/docapi_curl_get_all_docs_w_paging_1.result[]
----
--
====

> To get the next 5 documents re-run the same request, but add the `page-state`

Reviewer: In order to get the next five documents, re-run the request with `pageState` parameter set to the first page's `pageState`:

`/v2/namespaces/{namespace_name}/collections/{collections_name}` to retrieve
all the documents:

[tabs]
====
cURL command (/v2)::
+
--
[source,shell]
----
include::example$docapi/curl_get_all_docs.sh[]
----
--

Result::
+
--
[source,plaintext, subs="attributes+"]
----
include::example$result/docapi_curl_get_all_docs.result[]
----
--
====

The `page-size` parameter is included to get all the documents, rather than the
last inserted document.

==== Retrieving all documents with Paging
For large collections, you can page a subset of the results. In this example,
page-size is set to 5. Send a GET request to
/v2/namespaces/{namespace_name}/collections/{collections_name}page-size=5 to
retrieve the first five documents (a page) in the collection:

[tabs]
====
cURL command (/v2)::
+
--
[source,shell]
----
include::example$docapi/curl_get_all_docs_w_paging_1.sh[]
----
--

Result::
+
--
[source,plaintext, subs="attributes+"]
----
include::example$result/docapi_curl_get_all_docs_w_paging_1.result[]
----
--
====

Pay close attention to the `pageState` value in the results. The `pageState` is
a string representation a location in the result set. It is essentially an encoded

> shortcut which allows for the final result set to be built from a specific

Reviewer: ```suggestion
shortcut that allows the final result set to be built from a specific
```
